website/views/manage_carousel.py

```diff
@@ -0,0 +1,120 @@
+from django.shortcuts import render, redirect
+from django.contrib.auth.decorators import login_required
+from .tools import has_commit_permission
+from website.models import CarouselImage
+from website.forms import AddEditCarouselImageForm
+from django.http import Http404
+
+
+@login_required
+def dashboard_carousel(request):
+    try:
+        social = request.user.social_auth.get(provider='github')
+        access_token = social.extra_data['access_token']
+    except:
+        access_token = ''
+    has_permission = has_commit_permission(access_token, 'dipy_web')
+    if has_permission:
+        all_carousel_images = CarouselImage.objects.all()
+        context = {'all_carousel_images': all_carousel_images}
+        # if(request.user.has_perm('website.view_section')):
+        return render(request, 'website/dashboard_carousel.html', context)
+    else:
+        return render(request, 'website/dashboard_carousel.html', {})
+
+
+@login_required
+def add_carousel_image(request):
+    # check if user has edit permissions
+    try:
+        social = request.user.social_auth.get(provider='github')
+        access_token = social.extra_data['access_token']
+    except:
+        access_token = ''
+    if access_token:
+        has_permission = has_commit_permission(access_token, 'dipy_web')
+    else:
+        has_permission = False
+
+    # if user does not have edit permission:
+    if not has_permission:
+        return render(request, 'website/addeditcarousel.html', {})
+
+    # if user has edit permission:
+    context = {}
+    if request.method == 'POST':
+        submitted_form = AddEditCarouselImageForm(request.POST)
+        if submitted_form.is_valid():
+            submitted_form.save()
+            return redirect('/dashboard/carousel/')
+        else:
+            context['form'] = submitted_form
+            return render(request, 'website/addeditcarousel.html', context)
+
+    form = AddEditCarouselImageForm()
+    context['form'] = form
+    return render(request, 'website/addeditcarousel.html', context)
+
+
+@login_required
+def edit_carousel_image(request, carousel_image_id):
+    # check if user has edit permissions
```

Comment:
Actually checking permission using github API every time a view is loaded is not a good method either. I was thinking of making a mechanism of checking github permissions of users periodically and update their permissions accordingly in my database. I will open a new issue to keep track of it.